src/server/utils/bucket_diff.js

```diff
@@ -193,13 +202,36 @@ class BucketDiff {
         return list_objects_response.NextContinuationToken;
     }
 
+    /**
+     * @param {import("aws-sdk/lib/request").PromiseResult<AWS.S3.ListObjectVersionsOutput, AWS.AWSError> |
+    *         import("aws-sdk/lib/request").PromiseResult<AWS.S3.ListObjectsV2Output, AWS.AWSError>} list
+    *
+    * _get_delete_markers will return the delete markers grouped by key
+    *
+    */
+    _get_delete_markers(list_objects_response) {
```

Comment:
We should do a similar behaviour like `_object_grouped_by_key_and_omitted`
We should group the delete markers per key and omit when needed.

Reply:
Done, Thanks.